### PR TITLE
Run simulation mode as part of pfcp agent

### PR DESCRIPTION
**Pull request by pudelkoM**
Right now we have the spin up another pfcp agent instance to simulate session creation (`-simulate create`) or deletion. As this runs in a completely separate process from the persistent agent, they do not share any state. This is problematic when other features, like flow measure, require data from inserted PDRs. 
Realistically, production setups should not be affected as all programming happens through PFCP messages, but for local testing this is a huge shortcoming. Especially since there is no equivalent simulator on the PFCP level available.

This change adds a new `-simulate` flag value, `create_continue`, which runs the sim function as part of the agent startup procedure.

### CODE REVIEW

File: pfcpiface/main.go [hunk 1]
```diff
@@ -90,6 +92,43 @@ type IfaceType struct {
...
+}
+
+func (s *simMode) keepGoing() bool {
+	return strings.Contains(string(*s), "continue")
+}
+
+func (s *simMode) enable() bool {
+	return string(*s) != ""
+}
+
```
Comment (krsna1729) [inline pfcpiface/main.go:131]: Move into grpcsim.go
Use enum type for simMode to do string compare only once during Set

**Comment by krsna1729** [comment]
>  Especially since there is no equivalent simulator on the PFCP level available.

Can you please extend pfcpsim in a follow-up PR to do multi-session testing like grpcsim. This will be helpful for performance benchmarking of PFCP layer

**Comment by pudelkoM** [comment]
> Can you please extend pfcpsim in a follow-up PR to do multi-session testing like grpcsim. This will be helpful for performance benchmarking of PFCP layer

I can certainly try. It would be really nice to have, I agree.

File: pfcpiface/main.go (hunk 1 above)
Comment (pudelkoM) [reply pfcpiface/main.go:131]: Makes sense, the string type was just for demonstration purposes.

**Review by krsna1729 — APPROVED** [review]

**Comment by krsna1729** [comment]
retest this please

**Pull request merged**